From: Braving Authors <support@braving.org>
Date: Fri, 5 Sep 2026 10:17:00 +0000
Subject: [PATCH] [f-droid] Gate proprietary Google Play dependencies

Declare enable_play_services (default false in Braving) to exclude closed-source
Google Play libraries (installreferrer, play-app-update, play-integrity,
safetynet, play-review, billingclient) from brave_chrome_java_deps.
---
 build/android/config.gni | 24 ++++++++++++++++++++----
 1 file changed, 20 insertions(+), 4 deletions(-)

diff --git a/build/android/config.gni b/build/android/config.gni
index f33affe..4447b05 100644
--- a/build/android/config.gni
+++ b/build/android/config.gni
@@ -20,6 +20,10 @@ brave_android_manifest_includes = [
   "//brave/android/java/AndroidManifest_intent_filters.xml",
 ]
 
+declare_args() {
+  enable_play_services = false
+}
+
 brave_chrome_java_deps = [
   "//brave/android:qrreader_java",
   "//brave/android/java/org/chromium/chrome/browser/accessibility:java",
@@ -71,17 +75,12 @@ brave_chrome_java_deps = [
   "//brave/components/variations/android:java",
   "//brave/components/version_info/android:java",
   "//brave/components/webcompat_reporter/common:mojom_java",
-  "//brave/third_party/android_deps:com_android_installreferrer_java",
   "//brave/third_party/android_deps:com_github_bumptech_glide_java",
-  "//brave/third_party/android_deps:com_google_android_play_app_update_java",
-  "//brave/third_party/android_deps:com_google_android_play_integrity_java",
   "//brave/third_party/android_deps:com_google_code_gson_gson_java",
   "//brave/third_party/android_deps:com_wireguard_android_J__unpack_aar",
   "//brave/third_party/android_deps:com_wireguard_android_java",
-  "//brave/third_party/android_deps:google_play_services_safetynet_java",
   "//brave/third_party/zxing:zxing_java",
   "//brave/ui/android:base_java",
-  "//third_party/android_deps:com_google_android_play_review_java",
   "//third_party/android_deps:google_play_services_vision_common_java",
   "//third_party/android_deps:google_play_services_vision_java",
   "//third_party/android_deps:org_jetbrains_kotlinx_kotlinx_coroutines_android_java",
@@ -92,7 +91,16 @@ brave_chrome_java_deps = [
   "//url/mojom:url_mojom_origin_java",
 ]
 
-brave_chrome_java_deps += brave_app_billing_deps
+if (enable_play_services) {
+  brave_chrome_java_deps += [
+    "//brave/third_party/android_deps:com_android_installreferrer_java",
+    "//brave/third_party/android_deps:com_google_android_play_app_update_java",
+    "//brave/third_party/android_deps:com_google_android_play_integrity_java",
+    "//brave/third_party/android_deps:google_play_services_safetynet_java",
+    "//third_party/android_deps:com_google_android_play_review_java",
+  ]
+  brave_chrome_java_deps += brave_app_billing_deps
+}
 brave_chrome_java_deps += brave_app_playlist_deps
 
 if (enable_brave_wallet) {
